tools/db/generate.go

```diff
@@ -125,8 +125,15 @@ func generateContracts(db *sql.DB) error {
 	for i := uint64(1); i <= contractCount; i++ {
 		nodeID := rnd(1, nodeCount)
 		state := "Deleted"
-		if nodeUP[nodeID] && flip(contractCreatedRatio) {
-			state = "Created"
+		if nodeUP[nodeID] {
+			if flip(contractCreatedRatio) {
+				state = "Created"
+			} else if flip(0.05) {
+				state = "GracePeriod"
+			}
+		}
+		if state != "Deleted" && (minContractHRU > nodesHRU[nodeID] || minContractMRU > nodesMRU[nodeID] || minContractSRU > nodesSRU[nodeID]) {
+			continue
```

Comment:
can you add i-- (to maintain the number of contracts to be equal to contractsCount)?